### ISSUE TITLE
"Rectangle with rotation" results in gaps between collision masks

### ISSUE THREAD

**Issue by johanneshoerner** [YoYoGames/GameMaker-Bugs#2667]
### Description

When using "Rectangle with rotation" type of collision mask for sprites, there is a gap between adjacent masks. Because of this, objects that check for pixel perfect collision with position_meeting can fall through these gaps. The sprites don't have to be rotated for this behaviour to appear, they just have to have the "Rectangle with rotation" type enabled. 

When using "Rectangle" type of collision mask, there is no gap between adjacent masks. This is the expected behaviour. 

I can circumvent this bug by having duplicate sprites, one with rotation, one without. I don't know if this behaviour is intended, it seems like a bug to me that should be fixed.

Here is an example project. I am using LTS version 2022.0.2.49, so I don't habe a "Contact Us" package, but the bug is very easy to spot and to reproduce:
[collision_check.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13853227/collision_check.zip)

### Expected Change

"Rectangle with rotation" should have the same behaviour as "Rectangle", not leaving a gap between adjacent hitboxes.

### Steps To Reproduce

1. Start GameMaker
2. Put two sprites with "Rectangle" hitboxes next to each other
3. Check for collision with position_meeting near the edge
4. Change the hitboxes to "Rectangle with rotation"
5. Check for collision again
6. In the first case, there is no gap. In the second case, there is.

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2022.0.2 (LTS)

### Which platform(s) are you seeing the problem on?

Windows

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [X] I have included a small sample project

**Comment by jackerley** [comment]
Hi can you please provide the source project (File->Export Project->yyz) rather than a built version of the project? This will enable us to see exactly what is going on?

**Comment by johanneshoerner** [comment]
Of course, sorry I mixed that up, here's the project file: [collision_check.zip](https://github.com/YoYoGames/GameMaker-Bugs/files/13892618/collision_check.zip)
